src/rsqsim_api/rsqsim_api/catalogue/catalogue.py

```diff
@@ -463,12 +489,244 @@ def events_by_number(self, event_number: Union[int, np.int, Iterable[np.int]], f
+                                                facecolors=colour_dic[f_i],
+                                                cmap=log_cmap, norm=colors.LogNorm(vmin=log_min, vmax=log_max))
+                else:
+                    crustal_plot = ax.tripcolor(fault.vertices[:, 0], fault.vertices[:, 1], fault.triangles,
+                                                facecolors=colour_dic[f_i],
+                                                cmap=crustal_cmap, vmin=0., vmax=max_slip)
+                plots.append(crustal_plot)
+
+        if any([subplots is None, isinstance(subplots, str)]):
+            if plot_log_scale:
+                if subduction_list:
+                    sub_cbar = fig.colorbar(subduction_plot, ax=ax)
+                    sub_cbar.set_label("Slip (m)")
+                elif crustal_plot is not None:
+                    crust_cbar = fig.colorbar(crustal_plot, ax=ax)
+                    crust_cbar.set_label("Slip (m)")
+            else:
+                if subduction_list:
+                    sub_cbar = fig.colorbar(subduction_plot, ax=ax)
+                    sub_cbar.set_label("Subduction slip (m)")
+                if crustal_plot is not None:
+                    crust_cbar = fig.colorbar(crustal_plot, ax=ax)
+                    crust_cbar.set_label("Slip (m)")
+
+        plot_coast(ax=ax)
+
+        if write is not None:
+            fig.savefig(write, dpi=300)
+            if show:
+                plt.show()
+            else:
+                plt.close(fig)
+
+        if show and subplots is None:
+            plt.show()
+
+        return plots
+
+    def plot_GR(self,fault_model: RsqSimMultiFault):
+        """
+        Plot Gutenburg-Richter distribution for a given catalogue.
+        y-axis: log(annual frequency of events with M>Mw)
+        x axis: Mw
+        """
+        
+        mag_freq: dict[int, float | Any] = {}
+        #find min magnitude of catalogue
+        mws = np.array([ev.mw for ev in self.all_events(fault_model)])
+        min_mag=np.floor(np.min(mws))
+
+        for mag in range(int(min_mag), 10):
+            filt_cat = self.filter_whole_catalogue(min_mw=mag)
+            nevents = len(filt_cat.all_events(fault_model))
+            times = np.array([ev.t0 for ev in filt_cat.all_events(fault_model)])
+            tot_time = (np.max(times) - np.min(times)) / seconds_per_year
+            freq = nevents / tot_time
+            mag_freq[mag] = freq
+        #convert to dataframe for plotting
+        mf_dict = pd.DataFrame.from_dict(mag_freq, orient='index', columns=['freq'])
+        mf_dict.reset_index(inplace=True)
+        mf_dict.rename(columns={"index": "mag"}, inplace=True)
+        #mf_dict["log_freq"] = np.log10(mf_dict["freq"])
+        #plot
+        ax=mf_dict.plot.scatter(x="mag", y="freq")
+        ax.set_yscale('log')
+        ax.set_xticks(np.arange(min_mag,10,1))
+        plt.xlabel("Mw")
+        plt.ylabel("# events with M>Mw per year")
+        plt.show()
```

Comment:
Make optional write and show arguments like for plot_accumulated_slip

Reply:
We will probably add a subplot argument in case we want to make multi-panel figures